File: src/se/fredsberg/game/dice/zombie/ZombieDieFactory.java
package se.fredsberg.game.dice.zombie;

import java.awt.Color;
import java.util.LinkedList;
import java.util.List;

public class ZombieDieFactory {

    public ZombieDie redZombieDie() {
        return new ZombieDie(Color.RED, getRedZombieSides());
    }

    public ZombieDie yellowZombieDie() {
        return new ZombieDie(Color.YELLOW, getYellowZombieSides());
    }

    public ZombieDie greenZombieDie() {
        return new ZombieDie(Color.GREEN, getGreenZombieSides());
    }

    private List<ZombieValue> getRedZombieSides() {
        return getZombieSides(1, 2, 3);
    }

    private List<ZombieValue> getYellowZombieSides() {
        return getZombieSides(2, 2, 2);
    }

    private List<ZombieValue> getGreenZombieSides() {
        return getZombieSides(3, 2, 1);
    }

    private List<ZombieValue> getZombieSides(int brains, int feet, int shoot) {
        List<ZombieValue> sides = new LinkedList<ZombieValue>();
        addZombieValues(sides, brains, feet, shoot);
        return sides;
    }

    private void addZombieValues(List<ZombieValue> sides, int brains, int feet, int shoot) {
        addZombieValue(sides, brains, ZombieValue.BRAIN);
        addZombieValue(sides, feet, ZombieValue.FOOTPRINTS);
        addZombieValue(sides, shoot, ZombieValue.SHOOTGUN);
    }

    private void addZombieValue(List<ZombieValue> sides, int count, ZombieValue value) {
        for (int i = 0; i < count; i++) {
            sides.add(value);
        }
    }
}

File: jssrc/ZombieDieFactory.js
/**
 *
 */

function ZombieDieFactory() {
	this.getRedZombieSides = function() {
		return [ZombieValue.values.BRAIN, 
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.SHOOTGUN,
		        ZombieValue.values.SHOOTGUN,
		        ZombieValue.values.SHOOTGUN];
	};

	this.getYellowZombieSides = function() {
		return [ZombieValue.values.BRAIN,
		        ZombieValue.values.BRAIN,
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.SHOOTGUN,
		        ZombieValue.values.SHOOTGUN];
	};

	this.getGreenZombieSides = function() {
		return [ZombieValue.values.BRAIN,
		        ZombieValue.values.BRAIN,
		        ZombieValue.values.BRAIN,
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.FOOTPRINTS,
		        ZombieValue.values.SHOOTGUN];
	};
};

ZombieDieFactory.prototype.redZombieDie = function() {
	return new ZombieDie(Color.colors.RED, this.getRedZombieSides());
};

ZombieDieFactory.prototype.yellowZombieDie = function() {
	return new ZombieDie(Color.colors.YELLOW, this.getYellowZombieSides());
};

ZombieDieFactory.prototype.greenZombieDie = function() {
	return new ZombieDie(Color.colors.GREEN, this.getGreenZombieSides());
};


